src/app/components/core/notifications/notification-modal/notification-modal.component.ts

```diff
@@ -0,0 +1,202 @@
+
+  rulePluginChangedConfig = [];
+  deliveryPluginChangedConfig = [];
+  notificationChangedConfig = [];
+
+  @ViewChild(AlertDialogComponent) child: AlertDialogComponent;
+  @ViewChild('notificationConfigView') viewConfigItemComponent: ViewConfigItemComponent;
+  @ViewChild('ruleConfigView') ruleConfigView: ViewConfigItemComponent;
+  @ViewChild('deliveryConfigView') deliveryConfigView: ViewConfigItemComponent;
+
+  constructor(private configService: ConfigurationService,
+    private alertService: AlertService,
+    private notificationService: NotificationsService,
+    public ngProgress: ProgressBarService) { }
+
+  ngOnInit() { }
+
+  ngOnChanges() {
+    if (this.notification !== undefined) {
+      this.getCategory();
+      this.getRuleConfiguration();
+      this.getDeliveryConfiguration();
+    }
+  }
+
+  public toggleModal(isOpen: Boolean) {
+    const activeFilterTab = <HTMLElement>document.getElementsByClassName('accordion is-active')[0];
+    if (activeFilterTab !== undefined) {
+      const activeContentBody = <HTMLElement>activeFilterTab.getElementsByClassName('card-content')[0];
+      activeFilterTab.classList.remove('is-active');
+      activeContentBody.hidden = true;
+    }
+
+    const modalWindow = <HTMLDivElement>document.getElementById('south-service-modal');
+    if (isOpen) {
+      // this.notify.emit(false);
+      // this.svcCheckbox.setValue(this.service['schedule_enabled']);
+      modalWindow.classList.add('is-active');
+      return;
+    }
+    // this.notify.emit(false);
+    // this.isAdvanceConfig = true;
+    // this.getAdvanceConfig(null);
+    modalWindow.classList.remove('is-active');
+  }
+
+  public getRuleConfiguration(): void {
+    const categoryValues = [];
+    const notificationName = this.notification['name'].substr(this.notification['name'].indexOf(' ') + 1);
+    this.configService.getCategory(`rule${notificationName}`).
+      subscribe(
+        (data) => {
+          if (!isEmpty(data)) {
+            categoryValues.push(data);
+            this.ruleConfiguration = { key: `rule${notificationName}`, value: categoryValues };
+            this.useRuleProxy = 'true';
+          }
+        },
+        error => {
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText, true);
+          }
+        });
+  }
+
+  public getDeliveryConfiguration(): void {
+    /** request started */
+    this.ngProgress.start();
+    const categoryValues = [];
+    const notificationName = this.notification['name'].substr(this.notification['name'].indexOf(' ') + 1);
+    this.configService.getCategory(`delivery${notificationName}`).
+      subscribe(
+        (data) => {
+          if (!isEmpty(data)) {
+            categoryValues.push(data);
+            this.deliveryConfiguration = { key: `delivery${notificationName}`, value: categoryValues };
+            this.useDeliveryProxy = 'true';
```

Comment:
do we set to false? I think its always true. Or we change it based on config data chnage for Delivery Channel